plaso/filters/parser_filter.py

```diff
@@ -181,33 +188,41 @@ def ExpandPresets(self, presets_manager, expression):
     return self._JoinExpression(excludes, includes)
 
   def SplitExpression(self, expression):
-    """Determines the excluded and included parsers from an expression string.
+    """Determines the excluded and included values from an expression string.
+
+    Note that presets will not be expanded.
 
     Args:
       expression (str): parser filter expression, where None represents all
           parsers and plugins.
 
-          The parser filter expression is a comma separated value string that
-          denotes a list of parser names to include and/or exclude. Each entry
-          can have the value of:
+          A parser filter expression is a comma separated value string that
+          denotes which parsers should be used. Each value can be either:
+
+          * The name of a preset (case sensitive), which is a predefined list of
+            parsers and/or plugins (see data/presets.yaml for the default
+            presets).
+          * The name of a parser (case insensitive), for example 'msiecf'.
+          * The name of a plugin, prefixed with the parser name and a '/', for
+            example 'sqlite/chrome_history'.
 
-          * An exact match of a list of parsers, or a preset (see
-            data/presets.yaml for the list of predefined presets).
-          * A name of a single parser (case insensitive), such as msiecf.
-          * A glob name for a single parser, such as '*msie*' (case
-            insensitive).
+          If the value begins with a '!' character the item will be excluded
+          from the set of enabled parsers and plugins, otherwise the value will
+          be included.
 
     Returns:
       tuple: contains:
 
-        excludes (dict[str, set[str]]): excluded parsers and plugins,
-            the dictionary contains the names of plugins per parser name,
-            where '*' represents the parser itself and all its plugins.
-            A parser name can also be the name of a preset.
-        includes (dict[str, set[str]]): included parsers and plugins,
-            the dictionary contains the names of plugins per parser name,
-            where '*' represents the parser itself and all its plugins.
-            A parser name can also be the name of a preset.
+        excludes (dict[str, set[str]]): excluded presets, plugins and presets.
+            Dictionary keys are preset and/or parser names, and values are
+            either a set of plugin names to enable for a parser, or a set
```

Comment:
to enable ? did you mean to disable?

Reply:
`either` so it is not technically correct, since this function allows a `preset_name: set(['*', 'bla'])` as value so maybe:

and values are a set of plugin names to enable for a parser. Note that for a preset the set is expected to be "set([*])" but this restriction is not enforced by this function.

Reply:
Changed this around some more, but there's a lot of complexity in what and why this method returns.